rfc/0003-tests-arch.md

```diff
@@ -0,0 +1,234 @@
+|              |                                  |
+| :----------- | :------------------------------- |
+| Feature Name | Kubewarden testing architecture  |
+| Start Date   | Dec 22th 2021                    |
+| Category     | QA                               |
+| RFC PR       | [fill this in after opening PR]  |
+
+# Summary
+[summary]: #summary
+
+Define how, where and whom will trigger, store and run tests for the Kubewarden
+stack.
+
+# Motivation
+[motivation]: #motivation
+
+Currently in the Kubewarden project we do not have comprehensive tests which checks
+all the project components together in a automate way. We need to create a testing
+architecture that ensure that all changes in any component are tested to work
+together with the other components of the project. This should happen in a
+automated way with the as lowest overhead for the developer as possible.
+
+Furthermore, we should provide the minimal infrastructure to run component specific
+tests isolated from the rest of the stack. Thus, allowing the developer run checks
+ensuring that their changes does not broke something.
+
+## Examples / User Stories
+[examples]: #examples
+
+As a developer, I want to run unit tests locally after some code change with
+simple commands.
+
+As a developer, I want to run integration tests to ensure the program continue to
+behave as expected locally with simple commands.
+
+As a developer, I want to run the end-to-end tests to ensure that my changes impacting
+multiple Kubewarden stack components does not break some feature.
+
+As a QA engineer, I want to ensure that developers changes do not break the Kubewarden
+features.
+
+As QA engineer, I want to ensure that I can add tests with no impact on the
+development cycle.
+
+As a Kubewarden maintainer, I want to run all important tests on PR to minimize
+the risk of introducing new bugs and regressions.
```

Comment:
I like every single user story here. I think they also cover all relevant cases, except one.

We also want to ensure, as "Kubewarden maintainers" that existing policies keep working on new releases (e2e test) as they are expected to without any modification.